includes/functions/functions_products.php

```diff
@@ -693,6 +697,8 @@ function zen_get_products_description($product_id, $language_id = 0)
                              FROM " . TABLE_PRODUCTS_DESCRIPTION . "
                              WHERE products_id = " . (int)$product_id . "
                              AND language_id = " . (int)$language_id, 1);
+//Allow observer to modify the description
+    $zco_notifier->notify('NOTIFY_GET_PRODUCTS_DESCRIPTION', $product_id, $product);
```

Comment:
i am always a fan of less database hits rather than more.

if you want an observer to change be able to change the description (because that is all that gets returned by this function), why not do it before the `$db->Execute` and then have an early return?  ie starting on line 695:
```
    $description = '';
    $zco_notifier->notify('NOTIFY_GET_PRODUCTS_DESCRIPTION', $product_id, $language_id, $description);
    if (!empty($description)) {
        return $description;
    }
```
(if you do not want the observer to be able to change the language_id, you can create an array for the first argument along with the product_id.)

also note that on line 690, you will also need to add zco_notifier to the global scope as well as on line 20.  (did not your IDE complain about that??)

Reply:
I'm _guessing_ that the change suggested would allow an observer to manipulate the description returned by the database query (I'm thinking like WP 'shortcodes').

Agreed that both functions need to add a `global $zco_notifier;`.